admin/add_copyright.py

```diff
@@ -106,36 +105,62 @@ def addCopyrightToPythonFile(filename, lines):
     return newLines
 
 
+def updateCopyrightString(filename, lines):
+    newLines = list([])
+
+    iterLines = iter(lines)
+    copyrightReached = False
+    for line in iterLines:
+        newLines.append(line)
+        if line == "COPYRIGHT = (":
+            copyrightReached = True
+            break
+    if not copyrightReached:
+        raise RuntimeError(f"Start of copyright not found in {filename}.")
+
+    for line in COPYRIGHT:
+        newLines.append(f'   "{line}\\n"')
+
+    copyrightEndReached = False
+    for line in iterLines:
+        if line == ")":
+            newLines.append(line)
+            copyrightEndReached = True
+            break
+    if not copyrightEndReached:
+        raise RuntimeError(f"End of copyright not found in {filename}.")
+
+    for line in iterLines:
+        newLines.append(line)
+
+    return newLines
+
+
 def main():
     for root, dirs, files in os.walk(".", topdown=True):
-        for skip_dir in [".git", "reference"]:
-            if skip_dir in dirs:
-                dirs.remove(skip_dir)
+        for skip_dir in [
+            dir
+            for dir in dirs
+            if dir in [".git", "reference"] or dir.startswith(".nox")
+        ]:
+            dirs.remove(skip_dir)
```

Comment:
I don't know if the new version is clearer: a for-loop over a list comprehension seems a bit confusing.

The clearest alternative might be:

```python
def skip_dir(dir: str) -> bool:
    return dir in [".git", "reference"] or dir.startswith(".nox")

dirs[:] = [dir for dir in dirs if not skip_dir(dir)]
```

Reply:
Ok. The problem was that I don't want to add a second loop. I'll change it.

Reply:
Correct me if I'm wrong. The difference between `dir[:] = [...]` and `dir = [...]` is that the first change the elements in the list and the second one creates a new list.

Reply:
That is correct. Using a slice on the left hand side updates the existing `dirs` list. We must modify the existing `dirs` list object due to the way `os.walk()` works. Compare also our use of `os.walk()` in the [`gcovr.utils.search_file()` function](https://github.com/gcovr/gcovr/blob/master/gcovr/utils.py#L74-L82).